### PR TITLE
blocks: msg_pair_to_var: Log with block logger, not gr.log

### PR DESCRIPTION
**Pull request by marcusmueller**
This also adds the missing unit test.

Signed-off-by: Marcus Müller <mmueller@gnuradio.org>

## Description

The block was logging using gr.log.warn. It should have been logging using its own logger!

also, no unit tests makes me sad.

## Related Issue

#6799 looked at the block, and so I looked at the block, and was met with the above issues

## Which blocks/areas does this affect?

/

## Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md).
- [x] I have squashed my commits to have one significant change per commit. 
- [x] I [have signed my commits before making this PR](https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md#dco-signed)
- [x] My code follows the code style of this project. See [GREP1.md](https://github.com/gnuradio/greps/blob/main/grep-0001-coding-guidelines.md).
- [x] I have updated [the documentation](https://wiki.gnuradio.org/index.php/Main_Page#Documentation) where necessary.
- [x] I have added tests to cover my changes, and all previous tests pass.

### CODE REVIEW

**Comment by marcusmueller** [comment]
https://github.com/marcusmueller/gnuradio/actions/runs/6843384944/job/18605928276#step:6:9194

Only on windows this fails with:

```
   File "%SRC_DIR%\gr-blocks\python\blocks\qa_msg_pair_to_var.py", line 38, in test_calling
    test_msg = pmt.cons(pmt.intern("foo"), pmt.from_long(canary))
                                           ^^^^^^^^^^^^^^^^^^^^^
TypeError: from_long(): incompatible function arguments. The following argument types are supported:
    1. (x: int) -> pmt.pmt_python.pmt_base

Invoked with: 3736059631
```

uuuuuuuh, pretty sure 3736059631==0xdeadbeef is an integer value, and especially a valid python `int`. sure, it's larger than `1<<31`, but why is this failing here… will have to dig up what the correct behaviour is and why we're failing it (and whether PMT is missing a kind of obvious test case).

**Comment by marcusmueller** [comment]
To fix that error, for now, let's `from_uint64_t` instead :/

**Comment by marcusmueller** [comment]
Adding the don't merge until I squash (or squash yourself before merging, if you please!)

**Comment by willcode** [comment]
@marcusmueller Squashed/merged to main. Conda win is giving errors related to deadbeef. I've re-run the test a couple of times, now re-running all the Conda versions. Could you take a look and see if there's still a problem there?

File: gr-blocks/python/blocks/qa_msg_pair_to_var.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        def test_f(value: int = 0):
+            self.test_val = value
+
+        dut = blocks.msg_pair_to_var(test_f)
+        test_msg = pmt.cons(pmt.intern("foo"), pmt.from_uint64(canary))
+        src = blocks.message_strobe(test_msg, 10)
+        self.tb.msg_connect(*(src, pmt.intern("strobe")),
+                            *(dut, blocks.msg_pair_to_var.IN_PORT))
+        self.tb.start()
+        time.sleep(15e-3)
```
Comment (argilo) [inline gr-blocks/python/blocks/qa_msg_pair_to_var.py:43]: The short wait time here readily causes test failures on my system if I add some CPU load with `stress -c 8`.

File: gr-blocks/python/blocks/qa_msg_pair_to_var.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            if value == nightingale and value != lark:
+                raise Exception("It was the nightingale, and not the lark")
+
+        dut = blocks.msg_pair_to_var(test_f)
+        test_msg = pmt.cons(pmt.intern("birb"), pmt.from_uint64(nightingale))
+        src = blocks.message_strobe(test_msg, 10)
+        self.tb.msg_connect(*(src, pmt.intern("strobe")),
+                            *(dut, blocks.msg_pair_to_var.IN_PORT))
+        self.tb.start()
+        time.sleep(50e-3)
```
Comment (argilo) [inline gr-blocks/python/blocks/qa_msg_pair_to_var.py:65]: This also causes failures, although less frequently.

**Comment by argilo** [comment]
I opened #6981 to track the flaky tests.

**Pull request merged**